components/TextFields/src/MDCTextInputController.h

```diff
@@ -0,0 +1,204 @@
+
+ This property is modeled after the adjustsFontForContentSizeCategory property in the
+ UIConnectSizeCategoryAdjusting protocol added by Apple in iOS 10.0.
+
+ Default value is NO.
+ */
+@property(nonatomic, assign, readwrite, setter=mdc_setAdjustsFontForContentSizeCategory:)
+    BOOL mdc_adjustsFontForContentSizeCategory UI_APPEARANCE_SELECTOR;
+
+/** The behavioral style applied to the text input. */
+@property(nonatomic, assign)
+    MDCTextInputPresentationStyle presentationStyle NS_SWIFT_NAME(presentation);
+
+/** The text input the controller is affecting. */
+@property(nonatomic, nullable, strong) UIView<MDCTextInput> *textInput NS_SWIFT_NAME(input);
+
+/**
+ Controls when the underline will be shown.
+
+ The underline is an overlay that can be hidden depending on the editing state of the input text.
+
+ Default is UITextFieldViewModeAlways.
+ */
+@property(nonatomic, assign) UITextFieldViewMode underlineViewMode NS_SWIFT_NAME(underlineMode)
+    UI_APPEARANCE_SELECTOR;
+
+/**
+ Convenience init. Never fails.
+
+ @param input An MDCTextInput this controller will manage.
+ */
+- (nonnull instancetype)initWithTextInput:(nullable UIView<MDCTextInput> *)input
+    NS_SWIFT_NAME(init(input:));
+
+/**
+ Sets the state of the controller by setting the values of properties errorText and
+ errorAccessibilityValue.
+
+ The value of errorText controls the state of the text input.
+
+ @param errorText               The error text to be shown as underline text. (Copied.)
+ @param errorAccessibilityValue Optional override of default underline text accessibility value.
+                                (Copied.)
+
+ This method is usually called from whatever object is the UITextFieldDelegate for the MDCTextField.
+ That object is responsible for validation of the text and calling this method if the error state
+ needs to change.
+
+ When errorText != nil, the text input is in an error state:
+ - The error text appears in the underline text with the errorColor as text color.
+ - The input rectangle's underline, placeholder and character are colored to the errorColor.
+
+ When errorText == nil, the text input is not in error state:
+ - The underline text is restored to the color and value it was.
+ - The underline color and height is restored.
+ - The placeholder text is restored to the color it was.
+ - The character count text is restored to the color it was.
+
+ Setting errorText to an empty string (@"") will put the input in error state but the underline text
+ will be empty. It will color (if visible) the underline, the placeholder, and the character
+ count in the errorColor.
+
+ Setting errorAccessibilityValue when errorText == nil has no effect.
+
+ Note: The characterCountMax property also affects these same MDCTextInput properties.
+ */
+- (void)setErrorText:(nullable NSString *)errorText
+    errorAccessibilityValue:(nullable NSString *)errorAccessibilityValue
+    NS_SWIFT_NAME(set(errorText:errorAccessibilityValue:));
```

Comment:
This seems a bit off, closest thing that comes to mind is Dictionary:
[docs](https://developer.apple.com/reference/swift/dictionary/1539001-updatevalue)
`updateValue(_:forKey:)`
based on that I would be more inclined to expect something like `setErrorText(_,errorAccessibilityValue:)`
I'm not sure what the default here is.